File: report/src/main/java/com/vmlens/report/runelementtype/BarrierOperationType.java
package com.vmlens.report.runelementtype;

public enum BarrierOperationType {
    
    NOTIFY("Notify") , WAIT_ENTER("Wait Enter") , WAIT_EXIT("Wait Exit") ;

    private final String text;

    BarrierOperationType(String text) {
        this.text = text;
    }

    public String text() {
        return text;
    }
}

File: sync-bug/src/main/scala/com/anarsoft/race/detection/report/element/runelementtype/BarrierOperationType.scala
package com.anarsoft.race.detection.report.element.runelementtype

enum BarrierOperationType(val text : String) {
  case NOTIFY     extends BarrierOperationType("Notify")
  case WAIT_ENTER extends BarrierOperationType("Wait Enter")
  case WAIT_EXIT  extends BarrierOperationType("Wait Exit")
}